tests/publisher_test.py

```diff
@@ -104,6 +104,21 @@ def testJSONRecordPerLine(self):
                          result)
 
 
+class RemoteJSONPublisherTestCase(unittest.TestCase):
+
+  @mock.patch.object(publisher.RemoteJSONPublisher, 'PublishSamples')
```

Comment:
Instead of mocking RemoteJSONPublisher, I'd suggest mocking requests and then asserting that the right post function is made with the expected sample output.